### PR TITLE
Add some tests using buttercup

### PR DESCRIPTION
**Pull request by kermorgant**
first attempt at using buttercup for some basic tests

### CODE REVIEW

**Comment by kermorgant** [comment]
Hi @zonuexe 

I've written a pair of simple tests with buttercup. Travis would be nice to have that automated. Would it be ok for you to give travis access to this repository ?

**Comment by zonuexe** [comment]
@kermorgant Thank you and LGTM.

> Travis would be nice to have that automated. Would it be ok for you to give travis access to this repository ?

👍

**Comment by kermorgant** [comment]
@zonuexe fine. Then I think you need to setup Travis for emacs-php/phpactor.el as I don't seem to have permissions to do that.

**Comment by zonuexe** [comment]
@kermorgant 
I turned the repository on Travis CI.  I'm sorry for being late.

**Comment by kermorgant** [comment]
@zonuexe thanks !

**Comment by kermorgant** [comment]
hmm, looking at travis logs, it seems a package is missing.
any idea which one ?

**Comment by kermorgant** [comment]
Not tsure what's happening while testing if phpactor is installed correctly. Is there some async code issue ?

File: tests/e2e/test-sanity.el [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      (spy-on 'composer)
+      (setq composer--execute-interactive t
+            composer--async-use-compilation nil)
+      (setq return-value (phpactor-install-or-update))
+      ;; (sleep-for 15)
+      (expect 'composer :to-have-been-called)
+      (expect (spy-calls-args-for 'composer 0)
+            :to-equal
+            '(nil "install" "--no-dev"))
+      ;; (expect return-value :to-be "*compilation*")
```
Comment (kermorgant) [inline tests/e2e/test-sanity.el]: this test fails, couldn't figure out why

File: tests/e2e/test-sanity.el [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      ;; (expect install-result :to-be t)
+      ;; (display-warning 'buttercup (format "Buffer content : %s" (buffer-string* "*Messages*")))
+      ;; (dolist (bufname (buffer-list))
+      ;;   ;; (display-warning 'buttercup (format "found buffer : %s" bufname)
+      ;;   (display-warning 'buttercup (format "buffer content : %s %s" bufname (buffer-string* bufname)))
```
Comment (kermorgant) [inline tests/e2e/test-sanity.el]: There was no *compilation* buffer created

**Comment by kermorgant** [comment]
Hi @zonuexe 
 
I've started using buttercup and cask but it's quite tricky to figure out what's happening behind the scenes once the test is running. I would have liked to test the installation of phpactor from phpactor.el but for some unknown reason, the invocation of `composer install --no-dev` fails when phpactor-install-or-update is invoked from buttercup.

As you can see in .travis.yml, we could get around by running composer install directly, but that's not great. 

Do you have any idea about that ?

**Comment by kermorgant** [comment]
Hi @zonuexe 

I'm going to merge that in develop, but don't hesitate to add comments if you see something deserving a change.

**Pull request merged**